IPNI_HTTP_PROVIDER.md

```diff
@@ -0,0 +1,310 @@
+  provides a unified way of specifying network addresses, including information such as the network protocol, addressing
+  scheme, and transport options.
+
+These common data types play a crucial role in the IPNI HTTP Provider API specification by ensuring consistency and
+compatibility when working with content identification and cryptographic hash values. By utilizing these standardized
+data types, the API promotes interoperability and ease of integration within the IPNI ecosystem.
+
+### Encoding
+
+IPNI uses [InterPlanetary Data Model (IPLD)](https://ipld.io/docs/data-model/) to represent information associated to
+advertisements, their entries and their providers. The IPNI HTTP Provider API aims to offer a human-readable encoding
+for the request and response payloads. Therefore, the default encoding
+of [`application/vnd.ipld.dag-json`](https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-json) is
+used. However, implementers have the flexibility to specify different encodings by utilizing appropriate HTTP headers as
+long as it facilitates verifiability of the data exchanged. This allows for customization and optimization of the data
+encoding based on specific requirements or preferences. The choice of encoding should be communicated through `Accept`
+and `Content-Type` headers in the HTTP request and response headers. Alternative examples
+include [`application/vnd.ipld.dag-cbor`](https://www.iana.org/assignments/media-types/application/vnd.ipld.dag-cbor).
+
+As a special case, the following content types are interpreted as their IPLD equivelant:
+
+* `application/json` is considered to be the same as `application/vnd.ipld.dag-json`.
+* `application/cbor` is considered to be the same as `application/vnd.ipld.dag-cbor`.
+
+The common data types, such as `CID` and `Multihash`, are encoded in their
+standard [multibase format](https://github.com/multiformats/multibase) when referenced in URLs.
+
+### Versioning
+
+The IPNI Provider HTTP API adopts a versioning strategy that utilizes the path prefix to indicate major releases. As a
+best practice, future iterations that introduce breaking changes should increment the version number in the API path.
+This approach ensures compatibility and allows clients to adapt to new versions while maintaining backward compatibility
+with existing implementations. By explicitly incorporating versioning in the API path, it becomes easier to manage and
+track changes over time, facilitating smooth transitions and providing a clear indication of API evolution.
+
+
+### Namespacing
+
+This specification introduces the `/ipni/` URL prefix to namespace the functionality associated with IPNI HTTP providers. The presence of this namespace allows IPNI specifications to evolve without conflicting with other URL patterns that an HTTP server may support. Therefore, this specification reserves the right for any sub-tree that appears after this prefix.
```

Comment:
:thought_balloon:  Is the long term plan  to expose `/cid` and `/multihash`  to `/ipni/v1` namespace too?

Reply:
This is a provider spec only. the `cid` and `multihash` endpoints are for looking up records which reside on the server side and are outside of the scope of this doc.